tests/drivers/max31865/main.c

```diff
@@ -0,0 +1,157 @@
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the MAX31865 RTD-to-Digital converter driver
+ *
+ * @author      David Picard
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef SPI0_CS0
+/* use the default device and CS pin for ESP32 boards */
+#  define MAX31865_PARAM_PARAM_SPI    (SPI_DEV(0))
+#  define MAX31865_PARAM_PARAM_CS_PIN (SPI0_CS0)
+#else
+/* on non-ESP32 boards, the user must adapt the following 2 parameters */
+#  define MAX31865_PARAM_PARAM_SPI    (SPI_DEV(0))
+#  define MAX31865_PARAM_PARAM_CS_PIN GPIO_PIN(0, 5)
+#endif
+
+#include "max31865.h"
+#include "max31865_params.h"
+#include "ztimer.h"
+#include "saul_reg.h"
+#include "log.h"
+
+#define MEASUREMENT_SLEEP_MS        (1000)
+#define MAX_LOOPS_MEAS              (3)
+
+/* supported acquisition modes */
+enum {
+    TEST_MAX31865_NORMAL_MODE,
+    TEST_MAX31865_SAUL_MODE
+} acqmode = TEST_MAX31865_NORMAL_MODE;
+
+void print_fault(max31865_fault_t value);
+
+int main(void)
+{
+    max31865_t dev;
+    max31865_data_t data;
+    phydat_t phyTemp;
+    int cntMeas = 0;
+    saul_reg_t *saulDev = saul_reg;
+    phydat_t saulData;
+
+    puts("MAX31865 RTD-to-Digital converter test application\n");
+
+    /* initialize first configured sensor */
+    printf("%s() >> setting config byte: 0x%02X\n", __FUNCTION__, max31865_params[0].cfg_byte);
+    printf("Initializing MAX31865 converter...\t");
+    if (max31865_init(&dev, &max31865_params[0]) == 0) {
+        puts("[OK]\n");
+    }
+    else {
+        puts("[Failed]");
+        return 1;
+    }
+    uint8_t cfgreg = 0;
+    max31865_clear_fault(&dev, &max31865_params[0], &cfgreg);
+    LOG_DEBUG("%s() >> Config register readback = 0x%02X\n", __FUNCTION__, cfgreg);
+
+    puts("------------------------------");
+    puts("Switch to normal mode");
+
+    /* periodically convert temperature values */
+    while (1) {
+        ztimer_sleep(ZTIMER_MSEC, MEASUREMENT_SLEEP_MS);
+
+        switch (acqmode) {
+            case TEST_MAX31865_NORMAL_MODE:
+                max31865_read(&dev, &data);
+                phyTemp.val[0] = data.rtd_temperature_cdegc;    /* unit: 0.01°C */
+                phyTemp.scale = -2;             /* 1 cdegC = 1e-02 degC */
+                phyTemp.unit = UNIT_TEMP_C;     /* set the unit */
+                phydat_dump(&phyTemp, 1);       /* print the value in a pretty format */
+                break;
+            case TEST_MAX31865_SAUL_MODE:
+                while (saulDev) {
+                    saul_reg_read(saulDev, &saulData);
+                    phydat_dump(&saulData, 1);
+                    saulDev = saulDev->next;
+                }
+                saulDev = saul_reg; /* reset pointer for the next read */
+                break;
+            default:
+                puts("No mode selected. Do nothing.");
+                continue;
+        }
```

Comment:
We use the [Linux coding style](https://www.kernel.org/doc/Documentation/process/coding-style.rst), that is `switch` and `case` are aligned at the same indentation levels.
```suggestion
        switch (acqmode) {
        case TEST_MAX31865_NORMAL_MODE:
            max31865_read(&dev, &data);
            phyTemp.val[0] = data.rtd_temperature_cdegc;    /* unit: 0.01°C */
            phyTemp.scale = -2;             /* 1 cdegC = 1e-02 degC */
            phyTemp.unit = UNIT_TEMP_C;     /* set the unit */
            phydat_dump(&phyTemp, 1);       /* print the value in a pretty format */
            break;
        case TEST_MAX31865_SAUL_MODE:
            while (saulDev) {
                saul_reg_read(saulDev, &saulData);
                phydat_dump(&saulData, 1);
                saulDev = saulDev->next;
            }
            saulDev = saul_reg; /* reset pointer for the next read */
            break;
        default:
            puts("No mode selected. Do nothing.");
            continue;
        }
```

Reply:
Why did you mark this change request to resolved without any comment?

Reply:
Because I applied the change locally but it was too late yesterday night to push it. When am I supposed to mark the conversation resolved?

Reply:
Not relevant any more